pkg/adapter/apiserver/stats_reporter.go

```diff
@@ -0,0 +1,165 @@
+}
+
+var _ StatsReporter = (*reporter)(nil)
+
+// reporter holds cached metric objects to report filter metrics.
+type reporter struct {
+	namespaceTagKey             tag.Key
+	eventTypeTagKey             tag.Key
+	eventSourceTagKey           tag.Key
+	importerNameTagKey          tag.Key
+	importerResourceGroupTagKey tag.Key
+	responseCodeKey             tag.Key
+	responseCodeClassKey        tag.Key
+	filterResultKey             tag.Key
+}
+
+// NewStatsReporter creates a reporter that collects and reports apiserversource
+// metrics.
+func NewStatsReporter() (StatsReporter, error) {
+	var r = &reporter{}
+
+	// Create the tag keys that will be used to add tags to our measurements.
+	nsTag, err := tag.NewKey(metricskey.LabelNamespaceName)
+	if err != nil {
+		return nil, err
+	}
+	r.namespaceTagKey = nsTag
+
+	eventTypeTag, err := tag.NewKey(metricskey.LabelEventType)
+	if err != nil {
+		return nil, err
+	}
+	r.eventTypeTagKey = eventTypeTag
+
+	eventSourceTag, err := tag.NewKey(metricskey.LabelEventSource)
+	if err != nil {
+		return nil, err
+	}
+	r.eventSourceTagKey = eventSourceTag
+
+	importerNameTag, err := tag.NewKey(metricskey.LabelImporterName)
+	if err != nil {
+		return nil, err
+	}
+	r.importerNameTagKey = importerNameTag
+
+	importerResourceGroupTag, err := tag.NewKey(metricskey.LabelImporterResourceGroup)
+	if err != nil {
+		return nil, err
+	}
+	r.importerResourceGroupTagKey = importerResourceGroupTag
+
+	filterResultTag, err := tag.NewKey(LabelFilterResult)
+	if err != nil {
+		return nil, err
+	}
+	r.filterResultKey = filterResultTag
+	responseCodeTag, err := tag.NewKey(LabelResponseCode)
+	if err != nil {
+		return nil, err
+	}
+	r.responseCodeKey = responseCodeTag
+	responseCodeClassTag, err := tag.NewKey(LabelResponseCodeClass)
+	if err != nil {
+		return nil, err
+	}
+	r.responseCodeClassKey = responseCodeClassTag
+
+	// Create view to see our measurements.
+	err = view.Register(
+		&view.View{
+			Description: eventCountM.Description(),
+			Measure:     eventCountM,
+			Aggregation: view.Count(),
+			TagKeys: []tag.Key{r.namespaceTagKey, r.eventSourceTagKey,
+				r.eventTypeTagKey, r.importerNameTagKey,
+				r.importerResourceGroupTagKey, r.responseCodeKey,
+				r.responseCodeClassKey},
+		},
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	return r, nil
+}
+
+// ReportEventCount captures the event count.
+func (r *reporter) ReportEventCount(args *ReportArgs, responseCode int) error {
+	ctx, err := r.generateTag(args, responseCode)
+	if err != nil {
+		return err
+	}
+	metrics.Record(ctx, eventCountM.M(1))
+	return nil
+}
+
+func (r *reporter) generateTag(args *ReportArgs, responseCode int) (context.Context, error) {
+	return tag.New(
+		context.Background(),
+		tag.Insert(r.namespaceTagKey, args.ns),
+		tag.Insert(r.eventSourceTagKey, args.eventSource),
+		tag.Insert(r.eventTypeTagKey, args.eventType),
+		tag.Insert(r.responseCodeKey, strconv.Itoa(responseCode)),
+		tag.Insert(r.importerResourceGroupTagKey,
```

Comment:
same line... and let's keep the same order as when we created the view...

```go
return tag.New(
        context.Background(),
        tag.Insert(r.namespaceTagKey, args.ns),
        tag.Insert(r.eventSourceTagKey, args.eventSource),
        tag.Insert(r.eventTypeTagKey, args.eventType),
        tag.Insert(r.importerNameTagKey, args.name),
        tag.Insert(r.importerResourceGroupTagKey, importerResourceGroupValue),
        tag.Insert(r.responseCodeKey, strconv.Itoa(responseCode)),
        tag.Insert(r.responseCodeClassKey, utils.ResponseCodeClass(responseCode)))
```

Reply:
fixed